megatron/core/datasets/blended_megatron_dataset_builder.py

```diff
@@ -450,6 +452,30 @@ def _build_megatron_dataset_splits(
                     torch.distributed.barrier()
             return [None] * len(Split)
 
+        if self.cls == GPTDatasetFolder:
+            if not os.path.isdir(dataset_path):
+                raise RuntimeError(f"Not a directory: {dataset_path}")
+
+            mid_level_datasets = []
+            for i, _split in enumerate(Split):
+                if split[i] is None:
+                    mid_level_datasets.append(None)
+                else:
+                    mid_level_datasets.append(
+                        self.build_generic_dataset(
+                            self.cls,
+                            self.is_built_on_rank,
+                            synchronize_ranks,
+                            None, # indexed_dataset (unused)
+                            dataset_path, # folder_path
+                            None, # indexed_indices (unused)
+                            sizes[i],
+                            _split,
+                            self.config,
+                        )
+                    )
```

Comment:
I know we're not really using splits in our runs but it was weird not respecting it. I am not 100% sure this would actually work if you used splits though.